docs/architecture/domain.md

```diff
@@ -0,0 +1,125 @@
+## Domain Model
+
+![Definition](assets/definition-model.png)
+![Runtime](assets/runtime-model.png)
+
+### Source Code
+
+Source Code is a specified GitHub repository and branch that contains a `screwdriver.yaml` and the code required to build, test, and publish your application.
+
+### Step
+
+A step is a named action that needs to be performed, usually a single shell command. If the command finishes with a non-zero exit code, the step is considered a failure.
+
+### Container
+
+A container runs [steps] in an isolated environment. This is done in order to test the compatibility of code running in different environments with different versions, without affecting other [builds] that may be running at the same time. This is implemented using Docker containers.
+
+### Job
+
+A job consists of executing multiple sequential [steps] inside a specified [container]. If any step in the series fails, then the entire job is considered failed and subsequent steps will be skipped (unless configured otherwise).
+
+Jobs work by checking out the [source code] to a specified commit, setting the desired environment variables, and executing the specified [steps].
+
+During the job, the executing [steps] share three pieces of context:
+
+ - Filesystem
+ - [Container]
+ - [Metadata]
+
+Jobs are started by changes in [source code], as a trigger from the [workflow], or by human decision.
```

Comment:
Jobs can be started automatically by changes made in the source code or triggered through the workflow. Jobs can also be started manually through the UI.